sycl/plugins/unified_runtime/adapters/level_zero/ur_level_zero.hpp

```diff
@@ -0,0 +1,217 @@
+      {ZE_RESULT_ERROR_OUT_OF_DEVICE_MEMORY,
+       ZER_RESULT_ERROR_OUT_OF_DEVICE_MEMORY},
+      {ZE_RESULT_ERROR_OUT_OF_HOST_MEMORY,
+       ZER_RESULT_ERROR_OUT_OF_HOST_MEMORY}};
+
+  auto It = ErrorMapping.find(ZeResult);
+  if (It == ErrorMapping.end()) {
+    return ZER_RESULT_ERROR_UNKNOWN;
+  }
+  return It->second;
+}
+
+// Controls Level Zero calls tracing.
+enum DebugLevel {
+  ZE_DEBUG_NONE = 0x0,
+  ZE_DEBUG_BASIC = 0x1,
+  ZE_DEBUG_VALIDATION = 0x2,
+  ZE_DEBUG_CALL_COUNT = 0x4,
+  ZE_DEBUG_ALL = -1
+};
+
+const int ZeDebug = [] {
+  const char *DebugMode = std::getenv("ZE_DEBUG");
+  return DebugMode ? std::atoi(DebugMode) : ZE_DEBUG_NONE;
+}();
+
+// Prints to stderr if ZE_DEBUG allows it
+void zePrint(const char *Format, ...);
+
+// This function will ensure compatibility with both Linux and Windows for
+// setting environment variables.
+bool setEnvVar(const char *name, const char *value);
+
+// Perform traced call to L0 without checking for errors
+#define ZE_CALL_NOCHECK(ZeName, ZeArgs)                                        \
+  ZeCall().doCall(ZeName ZeArgs, #ZeName, #ZeArgs, false)
+
+// Flags which tell whether various Level Zero extensions are available.
+static bool ZeDriverGlobalOffsetExtensionFound = false;
+static bool ZeDriverModuleProgramExtensionFound = false;
+
+struct _ur_level_zero_platform : public _ur_platform {
+  _ur_level_zero_platform(ze_driver_handle_t Driver) : ZeDriver{Driver} {}
+  // Performs initialization of a newly constructed PI platform.
+  zer_result_t initialize();
+
+  // Level Zero lacks the notion of a platform, but there is a driver, which is
+  // a pretty good fit to keep here.
+  ze_driver_handle_t ZeDriver;
+
+  // Cache versions info from zeDriverGetProperties.
+  std::string ZeDriverVersion;
+  std::string ZeDriverApiVersion;
+  ze_api_version_t ZeApiVersion;
+
+  // Cache driver extensions
+  std::unordered_map<std::string, uint32_t> zeDriverExtensionMap;
+};
+
+using ur_level_zero_platform = _ur_level_zero_platform *;
+
+class ZeUSMImportExtension {
+  // Pointers to functions that import/release host memory into USM
+  ze_result_t (*zexDriverImportExternalPointer)(ze_driver_handle_t hDriver,
+                                                void *, size_t);
+  ze_result_t (*zexDriverReleaseImportedPointer)(ze_driver_handle_t, void *);
+
+public:
+  // Whether user has requested Import/Release, and platform supports it.
+  bool Enabled;
+
+  ZeUSMImportExtension() : Enabled{false} {}
+
+  void setZeUSMImport(ur_level_zero_platform Platform) {
+    // Whether env var SYCL_USM_HOSTPTR_IMPORT has been set requesting
+    // host ptr import during buffer creation.
+    const char *USMHostPtrImportStr = std::getenv("SYCL_USM_HOSTPTR_IMPORT");
+    if (!USMHostPtrImportStr || std::atoi(USMHostPtrImportStr) == 0)
+      return;
+
+    // Check if USM hostptr import feature is available.
+    ze_driver_handle_t driverHandle = Platform->ZeDriver;

```

Comment:
Minor.
Since this is kind of refactoring, suggest suggest defining(if not yet) coding guidelines for the unified runtime. Also, it would be nice if the same style is used in, at least, the same file. For example, here we have a mix of variables and functions starting with a capital and small letters.

Reply:
The code is supposed to follow intel/llvm coding standards. I have fixed the few violations that I noticed.
Note that this code is using L0 API which has different coding standards, so it's not possible to have the same standards in the same file.